File: mar-08/minimal.c
/*
 * minimal.c
 *
 *  Created on: 01/05/2013
 *      Author: valle
 */

#if __APPLE__
    #include <OpenGL/gl.h>
    #include <OpenGL/glu.h>
    #include <GLUT/glut.h>
#else
    #include <GL/gl.h>
    #include <GL/glu.h>
    #include <GL/glut.h>
#endif


static int camera = 0;


void init(void) {
    glClearColor(0.0, 0.0, 0.0, 0.0);
    glShadeModel(GL_FLAT);
}


void display(void) {
    glClear(GL_COLOR_BUFFER_BIT);
    glPushMatrix();
    glRotatef((GLfloat) camera, 1.0, 0.0, 0.0);
    glColor3f(1.0f,1.0f,0.0f);     /* Essa cor tem que mudar ! */
    glutWireSphere(1.0, 20, 16);   /* Essa forma geometrica tem que mudar (nao pode ser uma esfera) ! */
    glPopMatrix();
    glutSwapBuffers();
}


void reshape(int w, int h) {
    glViewport(0, 0, (GLsizei) w, (GLsizei) h);
    glMatrixMode(GL_PROJECTION);
    glLoadIdentity();
    gluPerspective(60.0,(GLfloat) w/(GLfloat) h, 1.0, 20.0);
    glMatrixMode(GL_MODELVIEW);
    glLoadIdentity();
    gluLookAt(0.0, 0.0, 5.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0);
}


void keyboard(unsigned char key, int x, int y) {
    switch (key) {
    case 'c':
        camera = (camera + 10) % 360;
        break;
    case 'C':
        camera = (camera - 10) % 360;
        break;
    default:
        return;
    }
   glutPostRedisplay();
}


int main(int argc, char** argv) {
    glutInit(&argc, argv);
    glutInitDisplayMode(GLUT_DOUBLE | GLUT_RGB);

    glutInitWindowSize(500, 500);
    glutInitWindowPosition(100, 100);
    glutCreateWindow(argv[0]);

    init();

    glutDisplayFunc(display);
    glutReshapeFunc(reshape);
    glutKeyboardFunc(keyboard);

    glutMainLoop();

    return 0;
}

File: mar-08/minimal.py
from __future__ import division
from __future__ import print_function

import sys

import OpenGL.GL as gl
import OpenGL.GLU as glu
import OpenGL.GLUT as glut


def init() :
    gl.glClearColor(0.0, 0.0, 0.0, 0.0)
    gl.glShadeModel(gl.GL_FLAT)


def display() :
    global camera
    gl.glClear(gl.GL_COLOR_BUFFER_BIT)
    gl.glPushMatrix()
    gl.glRotatef(camera, 1.0, 0.0, 0.0)
    gl.glColor3f(0.3, 0.7, 0.25)         # toroide de cor verde
    glut.glutWireTorus(0.5, 1.0, 20, 50) # criacao de um toroide:
                                         #   0.5 raio interno
                                         #   1.0 raio externo
                                         #   20 lados para cada secao radial
                                         #   50 divisoes radiais para o toroide
    gl.glPopMatrix()
    glut.glutSwapBuffers()


def reshape(w, h) :
    gl.glViewport (0, 0, w, h)
    gl.glMatrixMode(gl.GL_PROJECTION)
    gl.glLoadIdentity ()

    # angulo de visao na direcao Y modificado de 60.0 para 55.0
    glu.gluPerspective(55.0, w/h, 1.0, 20.0)

    gl.glMatrixMode(gl.GL_MODELVIEW)
    gl.glLoadIdentity()

    # posicao do olho no eixo X modificada de 0.0 para 1.0
    # posicao do olho no eixo Y modificada de 0.0 para 2.0
    # posicao do olho no eixo Z modificada de 5.0 para 3.5
    glu.gluLookAt(1.0, 2.0, 3.5, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0)


def keyboard(key, x, y) :
    global camera
    if key == 'c' :
        camera = (camera + 10) % 360
    elif key == 'C' :
        camera = (camera - 10) % 360
    else :
        return
    glut.glutPostRedisplay()


def main() :
    _ = glut.glutInit(sys.argv)
    glut.glutInitDisplayMode(glut.GLUT_DOUBLE | glut.GLUT_RGB)

    glut.glutInitWindowSize(500, 500)
    glut.glutInitWindowPosition(100, 100)
    _ = glut.glutCreateWindow(sys.argv[0])

    init()

    global camera
    camera = 0
    _ = glut.glutDisplayFunc(display)
    _ = glut.glutReshapeFunc(reshape)
    _ = glut.glutKeyboardFunc(keyboard)

    glut.glutMainLoop()


if __name__ == "__main__" :
    main()
